Adyen/Components/Base/LoadingComponent.swift

```diff
@@ -0,0 +1,48 @@
+//
+// Copyright (c) 2021 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Foundation
+
+/// :nodoc:
+public protocol LoadingComponent {
+    /// :nodoc:
+    func startLoading(for component: PaymentComponent)
+
+    /// Stops any processing animation that the view controller is running.
+    ///
+    /// - Parameters:
+    ///   - success: Boolean indicating the component should go to a success or failure state.
+    ///   - completion: Completion block to be called when animations are finished.
+    func stopLoading(withSuccess success: Bool, completion: (() -> Void)?)
+}
+
+extension LoadingComponent {
```

Comment:
Do we really need a default implementation for this? It do nothing.
Better remove the extension and actually see where are using it :)

Reply:
removed the default implementation and made some more small refactoring of `LoadingComponent`.